core/templates/dev/head/components/forms/schema_editors/SchemaBasedListEditorDirective.js

```diff
@@ -135,19 +135,35 @@ oppia.directive('schemaBasedListEditor', [
             }
           };
 
+          var deleteEmptyElements = function() {
+            for (var i = 0; i < $scope.localValue.length - 1; i++) {
+              if ($scope.localValue[i].length === 0) {
+                $scope.deleteElement(i);
+                i--;
+              }
+            }
+          };
+
           $scope.lastElementOnBlur = function() {
             _deleteLastElementIfUndefined();
             $scope.showAddItemButton();
           };
 
           $scope.showAddItemButton = function() {
+            deleteEmptyElements();
             $scope.isAddItemButtonPresent = true;
           };
 
           $scope.hideAddItemButton = function() {
             $scope.isAddItemButtonPresent = false;
           };
 
+          if ($scope.itemSchema().value === 'setInput') {
```

Comment:
Hey Sean!
The other way around that I can think of is we can use $scope.emit in the SchemaBasedListEditorDirective.js which emits the localValue to the parent setInput directive because to add an input field by default it has to be added to the localValue array also.